### PR TITLE
Add sample notebook to demo weaviate memory store

### PR DESCRIPTION
**Pull request by hsm207**
### Motivation and Context

  1. Why is this change required?
      To show users how to use the WeaviateMemoryStore memory store
  2. What problem does it solve?
      Users don't know how to config and use the WeaviateMemoryStore memory store
  3. What scenario does it contribute to?
      Users want to learn how to config and use the WeaviateMemoryStore memory store
  4. If it fixes an open issue, please link to the issue here.
      NA

### Description

Add a notebook to show users how to config and use the WeaviateMemoryStore memory store in a few examples.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by alexchaomander** [comment]
@hsm207 Thank you for this PR! Can you create this under `samples/notebooks/python/third_party`

And rename this to `weaviate-persistent-memory.ipynb`?

We want to start highlighting examples coming from other open-source projects and put them in their own location.

@awharrison-28 @mkarle

**Comment by hsm207** [comment]
> @hsm207 Thank you for this PR! Can you create this under `samples/notebooks/python/third_party`
> 
> And rename this to `weaviate-persistent-memory.ipynb`?

@alexchaomander done!

**Comment by hsm207** [comment]
@alexchaomander @awharrison-28 this PR is ready for review

File: samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb
Comment (awharrison-28) [inline samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb:173]:
```suggestion
    "from semantic_kernel.connectors.memory import weaviate_memory_store\n",
```
This will be updated accordingly in the next release

**Review by awharrison-28 — CHANGES_REQUESTED** [review]
This is really cool and I got it up and running on Windows + Docker fairly quickly! 

In addition to the other comments:

Please add the following block in the setup code
```
!pip install semantic-kernel==0.3.0.dev0
!pip install weaviate-client
!pip install python-dotenv
```

**Comment by hsm207** [comment]
> Please add the following block in the setup code

done!

File: samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb
Comment (hsm207) [reply samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb:72]: @awharrison-28 how does a user run Docker Desktop as admin? Or, did you actually mean a user needs to open a terminal as an admin and then run `docker compose up -d`?

File: samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb
Comment (dandv) [reply samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb]: Good idea, implemented.

**Comment by hsm207** [comment]
@awharrison-28 @alexchaomander we've addressed all your feedback

File: samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb
Comment (awharrison-28) [reply samples/notebooks/python/third_party/weaviate-persistent-memory.ipynb:72]: image

**Review by awharrison-28 — APPROVED** [review]
LGTM

**Comment by hsm207** [comment]
@alexchaomander just pending your review 😀

**Review by alexchaomander — APPROVED** [review]
Looks good to me! Thanks for adding this example!

**Comment by hsm207** [comment]
@alexchaomander thanks for the review. does this still need someone else's approval? The status of the actions is "6 workflows awaiting approval, This workflow requires approval from a maintainer".

**Pull request merged**
